tests/foreman/cli/test_usergroup.py

```diff
@@ -313,78 +304,72 @@ def test_negative_automate_bz1437578(self):
+                    'user-group-id': function_user_group['id'],
                     'name': 'Domain Users',
                 }
             )
-            self.assertEqual(
+            assert (
                 'Could not create external user group: '
                 'Name is not found in the authentication source'
                 'Name Domain Users is a special group in AD.'
                 ' Unfortunately, we cannot obtain membership information'
-                ' from a LDAP search and therefore sync it.',
-                result,
+                ' from a LDAP search and therefore sync it.' == result
             )
 
 
 @pytest.mark.run_in_one_thread
-class FreeIPAUserGroupTestCase(CLITestCase):
+class TestFreeIPAUserGroup:
     """Implements FreeIPA LDAP feature tests for user groups in CLI."""
 
-    @classmethod
     @skip_if_not_set('ipa')
-    def setUpClass(cls):
+    @pytest.fixture(scope='module')
+    def module_creds(self):
+        ldap_user_name = settings.ipa.username_ipa
+        ldap_user_passwd = settings.ipa.password_ipa
+        return {'username': ldap_user_name, 'password': ldap_user_passwd}
+
+    @skip_if_not_set('ipa')
+    @pytest.fixture(scope='module')
+    def module_auth(self, module_creds):
         """Read settings and create LDAP auth source that can be re-used in
         tests."""
-        super().setUpClass()
-        cls.ldap_user_name = settings.ipa.username_ipa
-        cls.ldap_user_passwd = settings.ipa.password_ipa
-        cls.base_dn = settings.ipa.basedn_ipa
-        cls.group_base_dn = settings.ipa.grpbasedn_ipa
-        cls.ldap_hostname = settings.ipa.hostname_ipa
-        cls.auth = make_ldap_auth_source(
+        base_dn = settings.ipa.basedn_ipa
+        group_base_dn = settings.ipa.grpbasedn_ipa
+        ldap_hostname = settings.ipa.hostname_ipa
+        auth = make_ldap_auth_source(
             {
                 'name': gen_string('alpha'),
                 'onthefly-register': 'true',
-                'host': cls.ldap_hostname,
+                'host': ldap_hostname,
                 'server-type': LDAP_SERVER_TYPE['CLI']['ipa'],
                 'attr-login': LDAP_ATTR['login'],
                 'attr-firstname': LDAP_ATTR['firstname'],
                 'attr-lastname': LDAP_ATTR['surname'],
                 'attr-mail': LDAP_ATTR['mail'],
-                'account': cls.ldap_user_name,
-                'account-password': cls.ldap_user_passwd,
-                'base-dn': cls.base_dn,
-                'groups-base': cls.group_base_dn,
+                'account': module_creds['username'],
+                'account-password': module_creds['password'],
+                'base-dn': base_dn,
+                'groups-base': group_base_dn,
             }
         )
+        yield auth
+        LDAPAuthSource.delete({'id': auth['server']['id']})
 
-    def setUp(self):
+    @pytest.fixture(scope='function')
+    def function_user_group(self):
```

Comment:
This fixture is the same in both classes, consider moving it out of the class so that its not defined multiple times.